pkg/engine/lifecycletest/target_test.go

```diff
@@ -1044,7 +1044,7 @@ func TestCreateDuringTargetedUpdate_UntargetedCreateReferencedByChangedTarget(t
 		_, err = monitor.RegisterResource("pkgA:m:typA", "b", true, deploytest.ResourceOptions{
 			Dependencies: []resource.URN{resA.URN},
 		})
-		assert.NoError(t, err)
+		assert.ErrorContains(t, err, "resource monitor shut down while waiting on step's done channel")
```

Comment:
Previously we didn't hit these assertions because the call to `RegisterResource` would never return. Now that we properly close the source iterator, the resource monitor gets shutdown and the call returns with a shutdown error.